### ISSUE TITLE
Problem loading User font [Terminus (TTF)] in macOS Catalina

### ISSUE THREAD

**Issue by trey-pindrop** [alacritty/alacritty#3909]
I am trying to get Alacritty to load my "Terminus (TTF)" font loaded as a User font in ~/Library/Fonts (Font Book screenshot
<img width="423" alt="Screen Shot 2020-07-02 at 11 51 23" src="https://user-images.githubusercontent.com/17730305/86381289-29598580-bc5b-11ea-8bb0-8e61731ea087.png">).  I've tried various forms of the font specification:

Terminus
"Terminus"
Terminus (TTF)
"Terminus (TTF)"
Terminus TTF
"Terminus TTF"
TerminusTTF
"TerminusTTF"

If Alacritty loads at all, it appears to use the default (Menlo) font.  I've attached the `-vv` output when setting it to "Terminus (TTF)" (as instructed by @kchibisov in #alacritty@freenode).  
[alacritty_output.txt](https://github.com/alacritty/alacritty/files/4865420/alacritty_output.txt).

And here's my alacritty.yml
[alacritty_yml.txt](https://github.com/alacritty/alacritty/files/4865436/alacritty_yml.txt)

I unfortunately do not have administrative/root access to this Mac, therefore I had to install Terminus (TTF) in ~/Library/Fonts.  I was able to workaround Alacritty.app being unsigned by copying it to ~/Applications/Alacritty.app.  The font is listed in the font picker in iTerm2 as "Terminus (TTF)", and works just fine there.

### System

OS: macOS Catalina
Version: alacritty 0.4.3 (c723247) [installed via Homebrew]

### Logs

Font/Terminal size: `alacritty -vv`

**Comment by chrisduerr** [comment]
It might be interesting to see if other applications can manage to render this font, would you mind giving that a shot?

Also I'd assume you've already done this, but you usually need to restart applications before a new font can be used. Make sure you don't rely on live reload to update the font, that only works when it has been installed before Alacritty was started (at least that's the case on Linux). To be 1000% sure, I'd probably just give the whole system a quick reboot if possible.

**Comment by trey-pindrop** [comment]
@chrisduerr On the Mac I originally posted this problem from, I had been using Terminus TTF in my User|~/Library/Fonts/ directory for some time (on the order of years).  I was able to reproduce it on another Mac, where I could install the font to /Library/Fonts/.  Here's where I got the Terminus TTF font from:  https://files.ax86.net/terminus-ttf/

I had been using Terminus TTF with iTerm2 and Terminal.app for quite some time, and those programs don't have issues displaying that font.  Apparently I have to change some of the code in darwin/mod.rs for the warnings to display in the alacritty log when it falls back to the default font.

I was able to successfully load the Hack TTF font into User|~/Library/Fonts/, and it works quite well.  So I think it's a problem with the Terminus TTF font itself, alacritty doesn't seem to work with it.  Alacritty either silently ignores the font, or generates an error and won't launch (i.e., with `family:  Terminus (TTF)`).

**Comment by trey-pindrop** [comment]
I use the `terminus-font` package in Arch Linux, and specify the font like so on that OS:

```
family:  xos4 Terminus
```
That's not the TTF version of the font.  There don't seem to be instructions for building `terminus-font` on macOS, and I haven't figured out a workaround with that.

**Comment by trey-pindrop** [comment]
@chrisduerr I've demonstrated this on two different macOS Catalina MacBook Pros (both are mid-2015 15" Retina MBPs).  What do I need to gather from each of them to confirm this bug?  As far as I can tell it's with the Terminus (TTF) font itself, however Alacritty is consuming it doesn't work, and it is silently falling back to the default (Menlo) font, or doesn't start at all depending on how I specify the font in `.alacritty.yml`.

**Comment by chrisduerr** [comment]
>  As far as I can tell it's with the Terminus (TTF) font itself

The question would just be what about the Terminus font is creating problems. Alacritty doesn't have any issues with TTF fonts in general.

**Comment by trey-pindrop** [comment]
It's something about this Terminus font.  I tried all the versions of Terminus in the [nerd-fonts repository](https://github.com/ryanoasis/nerd-fonts/tree/master/patched-fonts/Terminus/terminus-ttf-4.40.1), where they've renamed the font to Terminess because of licensing issues.  The fonts show up OK in Font Book.app, and are usable by both iTerm2 and Terminal.app, but Alacritty doesn't log any errors or debug messages about the font when it falls back to the default (Menlo).  If I use a font name Alacritty *can't* use, Alacritty doesn't launch and displays the error message posted above.

If you could instruct me on how to modify `darwin/mod.rs` so it prints useful information about the font when trying to load it, maybe we'll get some information on why this particular font family doesn't work.  I know zero rust, so I can only guess on what needs to be done to get more information.

The Font Book.app has a feature whereby it can validate fonts.  The only warning I get on the Terminus/Terminess fonts I've validated are for the Bold Italic style, the `'name' table usability` attribute shows a warning, which looks like a minor problem.  

This font family works on Arch Linux, but I'm not using the TTF version of the font there.

**Comment by kchibisov** [comment]
>If you could instruct me on how to modify darwin/mod.rs so it prints useful information about the font when trying to load it, maybe we'll get some information on why this particular font family doesn't work. I know zero rust, so I can only guess on what needs to be done to get more information.

Just replace all lines like `log::warn!` or `log::error!` with `println!`.

**Comment by trey-pindrop** [comment]
@chrisduerr @kchibisov I enabled the logging line per @kchibisov's suggestion, it doesn't look like it provides much information:

> Created log file at "/Users/tblancher/Alacritty-36450.log"
> [2020-07-13 14:41:37.769530000] [DEBUG] Using environment locale: en_US.UTF-8
> [2020-07-13 14:41:37.769818000] [INFO] Welcome to Alacritty
> [2020-07-13 14:41:37.769860000] [INFO] Configuration loaded from "/Users/tblancher/.alacritty.yml"
> Unable to load specified font Terminess, falling back to Menlo
> [2020-07-13 14:41:37.823990000] [DEBUG] Estimated DPR: 1
> [2020-07-13 14:41:37.824084000] [DEBUG] Estimated Cell Size: 10 x 21
> [2020-07-13 14:41:37.824111000] [DEBUG] Estimated Dimensions: None
> [2020-07-13 14:41:37.891470000] [INFO] Device pixel ratio: 1
> [2020-07-13 14:41:37.894975000] [INFO] Initializing glyph cache...
> Unable to load specified font Terminess, falling back to Menlo
> [2020-07-13 14:41:37.906145000] [INFO] ... finished initializing glyph cache in 0.011081033s 

So it happens during glyph cache initialization.  I took the above from the `-vvv` option to Alacritty, and it doesn't appear to give a reason why it can't load the font.  

I've forked the `alacritty` project, and created my own `terminus_font_troubleshooting` branch so I could build Alacritty.app.  If there's anything else you might want me to include in `darwin/mod.rs` that could give us clues as to why this particular font family is failing, let me know.

**Comment by kchibisov** [comment]
Are you sure that you're editing the right config file? The font name was `Terminess`, not `Terminus (TTF)` your config file is in $`HOME/.alacritty.yml`.

**Comment by trey-pindrop** [comment]
@kchibisov As I mentioned in [a post above](https://github.com/alacritty/alacritty/issues/3909#issuecomment-656929211), I found another version of Terminus TTF from the nerd-fonts repository.  Due to SIL license guidelines, they renamed the font to Terminess to add their specific changes to it.

Neither the original Terminus TTF nor the nerd-fonts Terminess TTF fonts work in Alacritty.app.  They work in iTerm2, Terminal.app, and appear to be mostly OK in Font Book.app (see the minor issue Font Book reports in the post above).

**Comment by trey-pindrop** [comment]
nerd-fonts provides several different versions of Terminess TTF, but none are explicitly compatible with macOS.  I've tried all of the different nerd-fonts variants of Terminess TTF (including the Windows compatible versions), and every other version of Terminus TTF that I could find.  None of them work with Alacritty.app.

**Comment by tblancher** [comment]
I just revisited this on my personal MacBook Pro (same 2015 Retina model as my work MacBook Pro, both running Catalina).  On this system, I unzipped the TerminessTTF ZIP archive in the system-wide /Library/Fonts.  I was able to get the TerminessTTF font working by specifying the following:

```
family: TerminessTTF Nerd Font
style:  Medium
```
The trick to getting it working was to change the `style` to `Medium`, not `Regular`.  I made the same change on my work MacBook Pro (with the font archive unzipped to my local home directory ~/Library/Fonts/), and it now works.  

I consider this resolved, sorry for all of the confusion.
